pkg/cmd/adm/register_member.go

```diff
@@ -160,3 +214,261 @@ func downloadScript(term ioutils.Terminal) (*os.File, error) {
+		if tc.Spec.APIEndpoint == hostName {
+			clusters = append(clusters, tc)
+		}
+	}
+
+	return clusters, nil
+}
+
+type registerMemberData struct {
+	hostClusterClient       runtimeclient.Client
+	memberClusterClient     runtimeclient.Client
+	hostApiEndpoint         string
+	memberApiEndpoint       string
+	hostOperatorNamespace   string
+	memberOperatorNamespace string
+	args                    registerMemberArgs
+	waitForReadyTimeout     time.Duration
+}
+
+type registerMemberValidated struct {
+	registerMemberData
+	hostToolchainClusterName      string
+	memberToolchainClusterName    string
+	warnings                      []string
+	errors                        []string
+	memberToolchainClusterOrdinal int
+}
+
+func dataFromArgs(ctx *extendedCommandContext, args registerMemberArgs, waitForReadyTimeout time.Duration) (*registerMemberData, error) {
+	hostConfig, err := clientcmd.LoadFromFile(args.hostKubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	hostClientConfig, err := clientcmd.NewDefaultClientConfig(*hostConfig, nil).ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	hostClusterClient, err := ctx.NewClientFromRestConfig(hostClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	hostApiEndpoint := getServerAPIEndpoint(hostConfig)
+	hostOperatorNamespace := args.hostNamespace
+
+	memberConfig, err := clientcmd.LoadFromFile(args.memberKubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	memberClientConfig, err := clientcmd.NewDefaultClientConfig(*memberConfig, nil).ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	memberClusterClient, err := ctx.NewClientFromRestConfig(memberClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	memberApiEndpoint := getServerAPIEndpoint(memberConfig)
+	memberOperatorNamespace := args.memberNamespace
+
+	return &registerMemberData{
+		args:                    args,
+		hostApiEndpoint:         hostApiEndpoint,
+		memberApiEndpoint:       memberApiEndpoint,
+		hostOperatorNamespace:   hostOperatorNamespace,
+		memberOperatorNamespace: memberOperatorNamespace,
+		hostClusterClient:       hostClusterClient,
+		memberClusterClient:     memberClusterClient,
+		waitForReadyTimeout:     waitForReadyTimeout,
+	}, nil
+}
+
+func (d *registerMemberData) validate(ctx *extendedCommandContext) (*registerMemberValidated, error) {
+	hostToolchainClusterName, err := utils.GetToolchainClusterName(string(configuration.Host), d.hostApiEndpoint, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// figure out the name that will be given to our new ToolchainCluster representing the member in the host cluster.
+	// This is the same name that the add-cluster.sh script will deduce and use.
+	membersInHost, err := getToolchainClustersWithHostname(ctx, d.hostClusterClient, d.memberApiEndpoint, d.hostOperatorNamespace)
+	if err != nil {
+		return nil, err
+	}
+	var ord int
+	if d.args.memberOrdinal == nil {
+		ord = len(membersInHost)
+	} else {
+		ord = int(*d.args.memberOrdinal)
+	}
```

Comment:
Yep. My misunderstanding here was that I understood "register member" to mean "register new member" and therefore wanted to use the numerical suffix as a way to be able to "increment" the suffix in case of collisions.

As you mentioned above, it would be simpler for `register-member` to be idempotent and let the caller decide what the suffix should be. This will simplify the logic in the code at the expense of requiring the user to specify the suffix explicitly (if there are multiple members in a single cluster).